containers/datalab/content/run.sh

```diff
@@ -14,107 +14,24 @@
 '
 
-ERR_MALFORMED_GATEWAY=1
-ERR_LOGIN=2
-ERR_PROJECT_NOT_FOUND=3
-ERR_ZONE_NOT_FOUND=4
-ERR_INSTANCE_NOT_FOUND=5
-ERR_DEPLOY=6
-ERR_GCLOUD_SSH_FAILED=7
-ERR_TUNNEL_FAILED=8
-ERR_GATEWAY_FAILED=9
-ERR_TMP_NOT_WRITABLE=10
-
-run_login() {
-  local login_cmd=${1:-"gcloud auth login"}
-
-  USER_EMAIL=`gcloud auth list --format="value(account)"`
-  if [[ -z "${USER_EMAIL}" ]]; then
-    local failed_login=""
-    ${login_cmd} || failed_login="true"
-    if [[ -n "${failed_login}" ]]; then
-      echo "Failed to log in to gcloud"
-      exit "${ERR_LOGIN}"
-    fi
-  fi
-  USER_EMAIL=`gcloud auth list --format="value(account)"`
-}
-
-kill_container() {
-  echo "Failing with error code $1"
-  # Docker will keep running as long as any processes remain
-  # running, so we have to kill all of them (not just the first process)
-  for PID in `ps -o pid=`; do
-    # We have to make sure that the current process does not get
-    # killed until it has finished killing all the other processes  
-    if [ "$PID" != "$BASHPID" ]; then
-      kill -SIGKILL $PID 2>&1 > /dev/null
-    fi
-  done
-
-  # Finally, we kill the current process.
-  exit $1
-}
+GATEWAY_DEPRECATED_MSG='Running Datalab against a kernel gateway is no longer supported.
 
-setup_tunnel() {
-  project_id=$1
-  zone=$2
-  instance=$3
-  ssh_user=$4
-  echo "Will connect to the kernel gateway running on the GCE VM ${instance} as ${ssh_user}"
-  gcloud compute ssh --quiet \
-    --project "${project_id}" \
-    --zone "${zone}" \
-    --ssh-flag="-fNL" \
-    --ssh-flag="localhost:8082:localhost:8080" \
-    --ssh-key-file="/content/datalab/.config/.ssh/google_compute_engine" \
-    "${ssh_user}@${instance}" || kill_container "${ERR_GCLOUD_SSH_FAILED}"
-
-  # Test that we can actually call the gateway API via the SSH tunnel
-  tunnel_failed=""
-  curl -o /tmp/kernel_specs http://localhost:8082/api/kernelspecs 2>/dev/null || tunnel_failed="true"
-  if [[ "${tunnel_failed}" == true ]]; then
-    echo "Failed to set up the SSH tunnel to the VM ${instance}"
-    echo "If the VM was recently created, then it may still be starting up, and retrying the command may work."
-    kill_container "${ERR_TUNNEL_FAILED}"
-  fi
-
-  default_kernel_spec=`cat /tmp/kernel_specs | python -c $'import json\nprint json.loads(raw_input())["default"]'`
-  if [[ -z "${default_kernel_spec}" ]]; then
-    echo "Failed to verify that the kernel gateway is running"
-    echo "If the VM was recently created, then it may still be starting up, and retrying the command may work."
-    kill_container "${ERR_GATEWAY_FAILED}"
-  fi
-  echo "Successfully established SSH tunnel to ${instance}"
+Please either switch to running all of Datalab in a VM via the `datalab` command line tool,
+or continue to use the unsupported image gcr.io/cloud-datalab/datalab:local-20170224
+'
 
-  watch_and_restart_tunnel "${project_id}" "${zone}" "${instance}" "${ssh_user}" &
-}
+ERR_UNSUPPORTED_GATEWAY_OPTION=1
+ERR_TMP_NOT_WRITABLE=2
```

Comment:
No, there is nothing that reads it.

This is just for debugging purposes if something exits unexpectedly.